uap_tracker/event_publisher.py

```diff
@@ -13,9 +14,27 @@ def listeners(self):
         return self._listeners
 
     def publish_process_frame(self, video_tracker):
+        listener_threads = []
         for listener in self.listeners:
-            listener.trackers_updated_callback(video_tracker)
+            listener_thread = Thread(target=self._publish_process_frame_task, args=(listener, video_tracker))
```

Comment:
I think a better approach here would be to use a Pool as it's cleaner and doesn't require creating and destroying threads. Pool size can be on config, but default to cores*0.75 or similar.

Reply:
Yeah, that is fair enough. I need to read up a little bit on it and then will change my implementation to use a threadpool instead of threads.